### ISSUE TITLE
Add RACE variable to unit-tests in Makefile

### ISSUE THREAD

**Issue by ruflin** [elastic/beats#1074]
Currently in unit-tests the RACE variable is missing. If it is added, race conditions are found in libbeat. See http://build-eu-00.elastic.co/job/Beats-PR/746/console

The line should look as following:

```
.PHONY: unit-tests
unit-tests: prepare-tests
	$(GOPATH)/bin/gotestcover -tags=unit $(RACE) -coverprofile=${COVERAGE_DIR}/unit.cov  ${GOPACKAGES}
```

**Comment by tsg** [comment]
I remember we removed `-race` because of the high CPU overhead. Now that we have `make unit` we could add it back, the only thing I'm worried about is even slower runs on Travis :(.

**Comment by ruflin** [comment]
As it is a variable, we can configure if it is enabled or not. By default it is disabled. We already have it actually in `make unit` but also disabled by default.

**Comment by urso** [comment]
I think the plan was to enabled these on jenkins like always, but disable locally and for travis (as travis is relatively slow).

**Comment by ruflin** [comment]
@urso Agree. That is why I think we just forgot to add it for unit-tests.

As a reference: More details on the race condition: https://github.com/elastic/beats/issues/1079

**Comment by andrewkroh** [comment]
Jenkins should be running with [`RACE_DETECTOR=1`](https://github.com/elastic/beats/blob/master/libbeat/scripts/Makefile#L45). I think the unit tests just need the `$(RACE)` add as ruflin suggests.

**Comment by andrewkroh** [comment]
Fixed in #1085
